lib/private/Files/Conversion/ConversionManager.php

```diff
@@ -80,24 +92,36 @@ public function convert(File $file, string $targetMimeType, ?string $destination
 		$fileMimeType = $file->getMimetype();
 		$validProvider = $this->getValidProvider($fileMimeType, $targetMimeType);
 
+		$targetExtension = '';
+		foreach ($this->getProvidersForMime($fileMimeType) as $mimeProvider) {
+			if ($mimeProvider->getTo() === $targetMimeType) {
+				$targetExtension = $mimeProvider->getExtension();
+				break;
+			}
+		}
+
 		if ($validProvider !== null) {
 			$convertedFile = $validProvider->convertFile($file, $targetMimeType);
 
-			if ($destination !== null) {
-				$convertedFile = $this->writeToDestination($destination, $convertedFile);
-				return $convertedFile->getPath();
+			// If destination not provided, we use the same path
+			// as the original file, but with the new extension
+			if ($destination === null) {
+				$basename = pathinfo($file->getPath(), PATHINFO_FILENAME);
+				$parent = $file->getParent();
+				$destination = $parent->getFullPath($basename . '.' . $targetExtension);
 			}
 
-			$tmp = $this->tempManager->getTemporaryFile();
-			file_put_contents($tmp, $convertedFile);
-
-			return $tmp;
+			$convertedFile = $this->writeToDestination($destination, $convertedFile);
+			return $convertedFile->getPath();
 		}
 
 		throw new RuntimeException('Could not convert file');
 	}
 
-	public function getProviders(): array {
+	/**
+	 * @return IConversionProvider[]
```

Comment:
```suggestion
	 * @return list<IConversionProvider>
```
The `[]` syntax is dangerous because it can mean multiple things.